lib/adapters/class-searchpress.php

```diff
@@ -132,18 +132,30 @@ public function add_search_suggest_to_indexed_post_data( $data, $sp_post ) {
 
 	/**
 	 * A callback for the sp_search_query_args filter. Adds aggregations to the
-	 * request so that the response will include aggregation buckets.
+	 * request so that the response will include aggregation buckets, and
+	 * filters by any aggregations set.
 	 *
 	 * @param array $es_args The request args to be filtered.
 	 * @return array The filtered request args.
 	 */
 	public function add_aggs_to_es_query( $es_args ): array {
 		// Add our aggregations.
 		foreach ( $this->get_aggregations() as $aggregation ) {
+			// Add aggregations to the request so buckets will be returned with results.
 			$request = $aggregation->request();
 			if ( ! empty( $request ) ) {
 				$es_args['aggs'][ $aggregation->get_query_var() ] = $request;
 			}
+
+			// Add any set aggregations to the query so results are properly filtered.
+			$filter = $aggregation->filter();
+			if ( ! empty( $filter ) ) {
+				if ( isset( $es_args['query']['bool'] ) ) {
+					$es_args['query']['bool']['filter'] = array_merge( $es_args['query']['bool']['filter'] ?? [], $filter );
+				} elseif ( isset( $es_args['query']['function_score']['query']['bool'] ) ) {
+					$es_args['query']['function_score']['query']['bool']['filter'] = array_merge( $es_args['query']['function_score']['query']['bool']['filter'] ?? [], $filter );
+				}
```

Comment:
this is fine, though I think it could be simplified (and, unless I'm thinking about it backwards, faster, because only the filtered results get scored) to do...

```php
				} elseif ( isset( $es_args['query'] ) ) {
					$es_args['query'] = [ 'bool' => [ 'filter' => $filter, 'must' => [ $es_args['query'] ] ] ];
				}
```